src/accel/SetupOptions.hh

```diff
@@ -238,6 +239,15 @@ struct SetupOptions
     {
         return max_num_tracks > 0 && static_cast<bool>(make_along_step);
     }
+
+    //! Initialize runtime-dependent defaults
+    SetupOptions()
+    {
+        // Set the default number of tracks per stream on device or host
+        max_num_tracks = celeritas::Device::num_devices() ? 262144 : 1024;
```

Comment:
This will need to be handled in the `inp` stuff or deferred till setup time: since people sometimes have static global `SetupOptions`, possibly before CUDA/MPI are initialized, `Device` can't be used here. You could make `max_num_tracks` and `initializer_capacity` a `std::optional<size_type>` and then set defaults when loading them?

Reply:
Ah right, damn. I initially set them in the `inp` in `ProblemSetup`, but we also need `max_num_tracks` from `SetupOptions` in the `LocalTransporter`. I'll rework it...